shelley-ma/shelley-ma-test/src/Test/Cardano/Ledger/EraBuffet.hs

```diff
@@ -163,11 +168,61 @@ instance FromCBOR (Annotator TestTxBody) where fromCBOR = pure <$> fromCBOR
 -- how one may choose pieces from the buffet to make an Era.
 {-
 data TestEra
-
 instance Era TestEra where
   type Crypto TestEra = TestCrypto
-
 type instance Value TestEra = Coin
 type instance TxBody TestEra = MaryTxBody TestEra
 type instance Script TestEra = TestScript
--}
\ No newline at end of file
+-}
+
+-- ==========================================================================================
+-- I am not sure that this stuff belongs here. But it does have one thing in common with the
+-- other stuff here. Both groups have to be able to see all the source code of every Era. and
+-- that the tests of every Era (if they are Era generic) have to be able to see this code.
+
+-- =====================================
+-- EraGen instaces for the MaryEra
+-- =====================================
+
+instance (CryptoClass.Crypto c) => ScriptClass (MaryEra c) where
```

Comment:
These Mary and Allegra instances can move to the `shelley-ma-test package` (which references this one and can share what it needs to).